### ISSUE TITLE
Should asset catalog OMAS be explicitly excluding subtypes in search results?

### ISSUE THREAD

**Issue by cmgrote** [odpi/egeria#5460]
At the moment, the asset catalog OMAS search method (`searchByType`) appears to be intentionally excluding subtypes in its search results:

```
// filter out the entities that are retrieved with the enterprise search.
// metadataCollection.findEntitiesByProperty includes in the search the subtypes of the provided type
if (!typesFilter.contains(entityDetail.getType().getTypeDefGUID())) {
    continue;
}
```

This means that with the default search types used (if none are explicitly configured) of `GlossaryTerm`, `Asset` and `SchemaElement` it is probably only `GlossaryTerm` that will ever return any results (the others are unlikely to have instances that exist at these super-type levels, but rather at some sub-type level like `DataFile`, `CSVFile` or `RelationalColumn`).

This means that the default behaviour of the UI makes it basically impossible to ever see any search results -- despite the fact that the performance hit of finding the results (enterprise connector federating queries to the cohort and retrieving all those subtype results back) is already done at the point at which they're explicitly excluded.

Is this really the intended behaviour?  If so, why?

(^^ @lpalashevski as discussed)

**Comment by lpalashevski** [comment]
Right. This is intended behaviour but not from asset catalog perspective :) I agree that is limiting functionallity as it is now. It was added to enhance user experiance but it is done in a wrong way, the idea was to do some smart filtering in the backend but not affect the existing functionallity (that means do it on the UI API and not asset catalog). So it was added as an intermediate step (or quick fix) before the UI gets enahnced and we allow smarter filtering input (like have two options lists one for assets and the other for related subtypes or similar).

I am also labeling it as "bug". Lets discuss the solution on the call tomorrow.

**Comment by lpalashevski** [comment]
Addressed in #5486

**Comment by lpalashevski** [comment]
Fixed with #5486
